### PR TITLE
add mawk

### PR DESCRIPTION
**Pull request by demhademha**
I'm not sure about versioning

### CODE REVIEW

**Comment by demhademha** [comment]
This is done, I'm just unsure about versioning

File: mawk.mk [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	# mawk.mk Prep mawk
+	cp -a $(BUILD_STAGE)/mawk $(BUILD_DIST)
+	# mawk.mk Sign
+	$(call SIGN,mawk,general.xml)
+	# mawk.mk Make .debs
+	$(call PACK,mawk,DEB_MAWK_V)
+	# mawk.mk Build cleanup
+	rm -rf $(BUILD_DIST)/mawk
+.PHONY: mawk mawk-package
+
```
Comment (TheRealKeto) [inline mawk.mk:36]: Remove this line...

File: mawk.mk [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	rm -rf $(BUILD_DIST)/mawk
+	mkdir -p $(BUILD_DIST)/mawk
+	# mawk.mk Prep mawk
+	cp -a $(BUILD_STAGE)/mawk $(BUILD_DIST)
+	# mawk.mk Sign
+	$(call SIGN,mawk,general.xml)
+	# mawk.mk Make .debs
+	$(call PACK,mawk,DEB_MAWK_V)
+	# mawk.mk Build cleanup
+	rm -rf $(BUILD_DIST)/mawk
```
Comment (TheRealKeto) [inline mawk.mk:34]: Add spaces between comments; improves readability.

File: mawk.mk [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+mawk-setup: setup
+	wget -q -nc -P$(BUILD_SOURCE) http://deb.debian.org/debian/pool/main/m/mawk/mawk_$(MAWK_VERSION).$(MAWK_COMMIT).orig.tar.gz
+	$(call EXTRACT_TAR,mawk_$(MAWK_VERSION).$(MAWK_COMMIT).orig.tar.gz,mawk-$(MAWK_VERSION)-$(MAWK_COMMIT),mawk)
+ifneq ($(wildcard $(BUILD_WORK)/mawk/.build_complete),)
+mawk:
+	@echo "Using previously built mawk."
+else
+mawk: mawk-setup
+	cd $(BUILD_WORK)/mawk && ./configure -C \
+	$(DEFAULT_CONFIGURE_FLAGS)
```
Comment (TheRealKeto) [inline mawk.mk:17]: Indent this line.

File: mawk.mk [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	+$(MAKE) -C $(BUILD_WORK)/mawk
+	+$(MAKE) -C $(BUILD_WORK)/mawk install \
+                DESTDIR=$(BUILD_STAGE)/mawk
+	touch $(BUILD_WORK)/mawk/.build_complete
+endif
```
Comment (TheRealKeto) [inline mawk.mk:22]: Add a space after this line.

File: mawk.mk [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+$(error Use the main Makefile)
+endif
+SUBPROJECTS   += mawk
+MAWK_VERSION := 1.3.4
+MAWK_COMMIT := 20200120
+DEB_MAWK_V   ?= $(MAWK_VERSION)
+mawk-setup: setup
+	wget -q -nc -P$(BUILD_SOURCE) http://deb.debian.org/debian/pool/main/m/mawk/mawk_$(MAWK_VERSION).$(MAWK_COMMIT).orig.tar.gz
+	$(call EXTRACT_TAR,mawk_$(MAWK_VERSION).$(MAWK_COMMIT).orig.tar.gz,mawk-$(MAWK_VERSION)-$(MAWK_COMMIT),mawk)
+ifneq ($(wildcard $(BUILD_WORK)/mawk/.build_complete),)
```
Comment (TheRealKeto) [inline mawk.mk:11]: Add a space before this line.

File: mawk.mk [hunk 6]
```diff
@@ -0,0 +1,36 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
```
Comment (TheRealKeto) [inline mawk.mk:3]: Add a space after this line.

File: mawk.mk [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+SUBPROJECTS   += mawk
+MAWK_VERSION := 1.3.4
+MAWK_COMMIT := 20200120
+DEB_MAWK_V   ?= $(MAWK_VERSION)
+mawk-setup: setup
+	wget -q -nc -P$(BUILD_SOURCE) http://deb.debian.org/debian/pool/main/m/mawk/mawk_$(MAWK_VERSION).$(MAWK_COMMIT).orig.tar.gz
```
Comment (TheRealKeto) [inline mawk.mk:9]: Add a space after line 8. Add a space after the argument, -P; improves readability.

File: mawk.mk [hunk 8]
```diff
@@ -0,0 +1,36 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+SUBPROJECTS   += mawk
+MAWK_VERSION := 1.3.4
+MAWK_COMMIT := 20200120
+DEB_MAWK_V   ?= $(MAWK_VERSION)
```
Comment (TheRealKeto) [inline mawk.mk:7]: Fix the spacing of your vars.

File: mawk.mk [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+	mkdir -p $(BUILD_DIST)/mawk
+	# mawk.mk Prep mawk
+	cp -a $(BUILD_STAGE)/mawk $(BUILD_DIST)
+	# mawk.mk Sign
+	$(call SIGN,mawk,general.xml)
+	# mawk.mk Make .debs
+	$(call PACK,mawk,DEB_MAWK_V)
+	# mawk.mk Build cleanup
+	rm -rf $(BUILD_DIST)/mawk
+.PHONY: mawk mawk-package
```
Comment (TheRealKeto) [inline mawk.mk:35]: Add a space before this line.

File: mawk.mk [hunk 10]
```diff
@@ -0,0 +1,36 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+SUBPROJECTS   += mawk
+MAWK_VERSION := 1.3.4
+MAWK_COMMIT := 20200120
+DEB_MAWK_V   ?= $(MAWK_VERSION)
+mawk-setup: setup
```
Comment (TheRealKeto) [inline mawk.mk:8]: Add a space above this line.

File: build_info/mawk.control [hunk 11]
```diff
@@ -0,0 +1,19 @@
+Package: mawk
+Version: @DEB_MAWK_V@
+Architecture: @DEB_ARCH@
+Maintainer: @DEB_MAINTAINER@
+Section: Utilities
+Priority: required
```
Comment (TheRealKeto) [inline build_info/mawk.control:6]: Change this to optional.

**Review by TheRealKeto — CHANGES_REQUESTED** [review]
Please unmark this as a draft once the following changes have been made.

File: mawk.mk [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+	
+	# mawk.mk Sign
+	$(call SIGN,mawk,general.xml)
+	
+	# mawk.mk Make .debs
+	$(call PACK,mawk,DEB_MAWK_V)
+	
+	# mawk.mk Build cleanup
+	rm -rf $(BUILD_DIST)/mawk
+	.PHONY: mawk mawk-package
```
Comment (TheRealKeto) [inline mawk.mk:43]: De-dent and add a space before this line.

**Review by TheRealKeto — CHANGES_REQUESTED** [review]

**Comment by CRKatri** [comment]
Add the postinst and postrm (https://salsa.debian.org/debian/mawk/-/blob/master/debian/mawk.postinst) and do the same for gawk, and make them both provide the awk package.

**Pull request merged**